app/models/site_status.rb

```diff
@@ -43,4 +46,27 @@ class SiteStatus < ApplicationRecord
   }
   scope :with_vacancies, -> { where.not(vac_status: :no_vacancies) }
   scope :open_for_applications, -> { findable.applications_being_accepted_now.with_vacancies }
+
+private
+
+  def vac_status_must_be_consistent_with_course_study_mode
+    unless vac_status_consistent_with_course_study_mode?
+      errors.add(:vac_status, "(#{vac_status}) must be consistent with course study mode #{course.study_mode}")
+    end
+  end
+
+  def vac_status_consistent_with_course_study_mode?
```

Comment:
Nice! We’ve got this same method on my frontend PR so good to have here instead 👌

Reply:
Let's keep this for consistency then